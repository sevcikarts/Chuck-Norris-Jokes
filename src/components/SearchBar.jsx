import axios from "axios";
import React  from "react";
import TextField from "@material-ui/core/TextField";

const SearchBar = ({ query, setQuery, setRandomJoke, setError }) => {

  const handlechange = (e) => {
    e.preventDefault();
    setQuery(e.target.value);
  };

  const handleSearch = (e) => {
    e.preventDefault();
    if (query.length > 2) {
      setRandomJoke([""]);
      setError("");
      axios
        .get(`https://api.chucknorris.io/jokes/search?query=${query}`)
        .then((response) => {
          response.data.result.map((item) =>
            setRandomJoke((state) => [...state, item.value])
          );
          let total = response.data.total;

          if (total === 0) {
            setError("expression not found");
          }
        });
    } else {
      setError("enter at least 3 characters");
    }
  };
  return (
    
    <div>
     <form noValidate autoComplete="off" onSubmit={handleSearch}>
     <TextField
      onChange={handlechange}
      style={{ width: "250px" }}
      id="standard-basic"
      label="random by query"
      type="text"
      value={query}
    />
      </form>
    </div>
  );
};

export default SearchBar;
