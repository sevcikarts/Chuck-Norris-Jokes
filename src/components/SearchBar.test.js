import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import SearchBar from "./SearchBar.jsx";
import toJSON from "enzyme-to-json"
import TextField from "@material-ui/core/TextField";

Enzyme.configure({ adapter: new Adapter() });



describe("rendering", () => {
  it("rendering div", () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper.find("div").text()).toContain("");
  });

  it("rendering without crashing", () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper.find("div").text()).toHaveValue;
  });

  it("rendering TextField", () => {
    const wrapper = shallow(<SearchBar />);
    const input = wrapper.find(TextField);
    expect(input.exists()).toBe(true);
  });
  it("snapShot", () => {
    const wrapper = shallow(
      <SearchBar/>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("rendering TextField", () => {
    const wrapper = shallow(<SearchBar  />);
    const input = wrapper.find(TextField);
    expect(input.length).toBe(1);
  });
  

});

