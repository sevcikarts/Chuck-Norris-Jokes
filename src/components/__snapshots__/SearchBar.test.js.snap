// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering snapShot 1`] = `
<div
  className="inControll"
>
  <form
    autoComplete="off"
    noValidate={true}
    onSubmit={[Function]}
  >
    <WithStyles(ForwardRef(TextField))
      id="standard-basic"
      label="random by query"
      onChange={[Function]}
      style={
        Object {
          "width": "300px",
        }
      }
      type="text"
    />
  </form>
</div>
`;
